### DISCUSSION TITLE
建表问题 想用字段cola做唯一键 同时用时间字段colc做分区 应如何建表

### DISCUSSION THREAD

**Discussion by cdavis26424** [apache/doris#6822] [A - General / Q&A]
Doris version 0.12.0-rc03
不能同时用UNIQUE KEY和DUPLICATE KEY
create table test_doris (
  cola INT,
  colb DECIMAL(8, 3),
  colc DATETIME,
  cold VARCHAR(30),
  cole CHAR(3)
)
UNIQUE KEY(`cola`)
DUPLICATE KEY(`colc`)
partition by range (colc)()
DISTRIBUTED BY HASH(`cola`)
PROPERTIES
(
    "replication_num" = "3",
    "dynamic_partition.enable" = "true",
    "dynamic_partition.time_unit" = "MONTH",
    "dynamic_partition.end" = "3",
    "dynamic_partition.prefix" = "p",
    "dynamic_partition.buckets" = "16"
);

**Comment by hf200012** [comment]
create table test_doris (
cola INT,
colc DATETIME,
colb DECIMAL(8, 3),
cold VARCHAR(30),
cole CHAR(3)
)
UNIQUE KEY(cola,colc)
partition by range (colc)()
DISTRIBUTED BY HASH(cola)
PROPERTIES
(
"replication_num" = "3",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "MONTH",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "16"
);

**Reply by cdavis26424** [reply]
这样写只有cola,colc都相同才会认为已存在这行，而我需要只要cola相同就认为这行已存在
